classes/announcement/maps/Schema.php

```diff
@@ -97,10 +105,26 @@ protected function mapByProperties(array $props, Announcement $item): array
             }
         }
 
-        $output = $this->schemaService->addMissingMultilingualValues($this->schema, $output, $this->context->getSupportedFormLocales());
+        $output = $this->schemaService->addMissingMultilingualValues($this->schema, $output, $this->getSupportedLocales());
 
         ksort($output);
 
         return $this->withExtensions($output, $item);
     }
+
+    protected function getUrlPath(): string
```

Comment:
Reviewing this, I'm thinking it's time to start formalizing the "contract" that both context and site need to implement, rather than embedding conditional logic. I'd support the use of e.g. an interface on both `Site` and `Context`, if you didn't mind the additional bit of work. This has already come up several times recently for locales -- see https://github.com/pkp/pkp-lib/issues/8025.

I think an interface here would help clean up a lot of functions that risk getting duplicated/scattered around the codebase -- isAnnouncementsEnabled, getContextId, getAnnouncementsIntro, ...

Reply:
Makes sense, but this contract is on a pretty tight schedule. Can I avoid the rewrite for now and see what time I've got left later on?

Reply:
Yes, that's fine -- but this is the kind of thing that's harder to refactor later, so hoping there's room. I wouldn't expect the interface to go beyond strictly what you need here; our other uses for it would be on the dev team's plate.